gum/backend-qnx/gumprocess-qnx.c

```diff
@@ -175,30 +263,33 @@ gum_qnx_enumerate_ranges (pid_t pid,
   g_free (as_path);
 
   res = devctl (fd, DCMD_PROC_PAGEDATA, 0, 0, &num_mapinfos);
-  g_assert (res != 0);
+  g_assert (res == 0);
 
   mapinfos = g_malloc (sizeof (procfs_mapinfo) * num_mapinfos);
-  debuginfo = g_malloc (sizeof (procfs_debug) + 0x100);
+  debuginfo = g_malloc (sizeof (procfs_debuginfo) + 0x100);
 
-  res = devctl (fd, DCMD_PROC_PAGEDATA, &mapinfos,
-      sizeof (procfs_mapinfo) * num_mapinfos, NULL);
-  g_assert (res != 0);
+  res = devctl (fd, DCMD_PROC_PAGEDATA, mapinfos,
+      sizeof (procfs_mapinfo) * num_mapinfos, &num_mapinfos);
+  g_assert (res == 0);
 
   while (carry_on && i != num_mapinfos)
   {
     GumRangeDetails details;
     GumMemoryRange range;
+    GumFileMapping file;
 
     range.base_address = mapinfos[i].vaddr;
     range.size = mapinfos[i].size;
 
     details.range = &range;
-    details.prot = gum_page_protection_from_page_data_flags (mapinfos[i].flags);
+    details.prot = _gum_page_protection_from_posix (mapinfos[i].flags);
 
-    debuginfo.vaddr = mapinfos[i].vaddr;
+    debuginfo->vaddr = mapinfos[i].vaddr;
     res = devctl (fd, DCMD_PROC_MAPDEBUG, debuginfo,
         sizeof (procfs_debuginfo) + 0x100, NULL);
-    details.file = debuginfo.path;
+    g_assert (res == 0);
+    file.path = debuginfo->path;
+    details.file = &file;
```

Comment:
When possible it's preferable to order the initializations from highest to lowest level. So start with initializing the GumRangeDetails, then move on to the GumMemoryRange, then finally the GumFileMapping.